### PR TITLE
Upgrade to Polkadot v0.9.37

### PR DESCRIPTION
**Pull request by jasl**
https://github.com/paritytech/substrate/compare/polkadot-v0.9.34...polkadot-v0.9.35
https://github.com/paritytech/substrate/compare/polkadot-v0.9.35...polkadot-v0.9.36
https://github.com/paritytech/substrate/compare/polkadot-v0.9.36...polkadot-v0.9.37

Breaking changes for us:
- Update Subxt to latest master ( https://github.com/paritytech/subxt/commit/e4e9562b45451f807099a0749dfc4f9a100247c6 ) beyond 0.25
- https://github.com/paritytech/substrate/pull/12795
- https://github.com/paritytech/substrate/pull/12891
- https://github.com/paritytech/substrate/pull/12868
- https://github.com/paritytech/substrate/pull/12845
- https://github.com/paritytech/substrate/pull/12852
- https://github.com/paritytech/substrate/pull/12883
- https://github.com/paritytech/substrate/pull/12874

Special note for https://github.com/paritytech/substrate/pull/12891 in the future, we have to write `call_index` for every extrinsic calls, or it will fail to compilation

Other notable changes:
- https://github.com/paritytech/substrate/pull/12844
- https://github.com/paritytech/substrate/pull/12663
- https://github.com/paritytech/substrate/pull/12537
- https://github.com/paritytech/substrate/pull/12953
- https://github.com/paritytech/substrate/pull/13080

TODO: These Subxt change breaks Pherry
- https://github.com/paritytech/subxt/pull/691
- https://github.com/paritytech/subxt/pull/694
- https://github.com/paritytech/subxt/pull/760

### CODE REVIEW

**Comment by jasl** [comment]
There's an error, relates to https://github.com/paritytech/substrate/pull/12883

```
   Compiling pink v0.1.0 (/Users/jasl/Workspaces/BlockChain/phala-blockchain/crates/pink)
error[E0599]: the method `as_ref` exists for reference `&<<E as pallet_contracts::chain_extension::Ext>::T as SysConfig>::AccountId`, but its trait bounds were not satisfied
   --> crates/pink/src/runtime/extension.rs:105:14
    |
105 |             .as_ref()
    |              ^^^^^^ method cannot be called on `&<<E as pallet_contracts::chain_extension::Ext>::T as SysConfig>::AccountId` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
            `<<E as pallet_contracts::chain_extension::Ext>::T as SysConfig>::AccountId: AsRef<_>`
            which is required by `&<<E as pallet_contracts::chain_extension::Ext>::T as SysConfig>::AccountId: AsRef<_>`

For more information about this error, try `rustc --explain E0599`.
```

File: crates/phactory/Cargo.toml [hunk 1]
```diff
@@ -34,11 +34,11 @@ phala-crypto = { path = "../phala-crypto", features = ["getrandom", "stream"] }
...
-sp-runtime-interface = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.34", features = ["disable_target_static_assertions"] }
-sp-runtime           = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.34" }
-sp-externalities     = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.34" }
-sp-state-machine     = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.34" }
+sp-io                = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36", features = ["disable_panic_handler", "disable_oom", "disable_allocator"] }
+sp-runtime-interface = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36", features = ["disable_target_static_assertions"] }
+sp-runtime           = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
+sp-externalities     = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
+sp-state-machine     = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36" }
```
Comment (kvinwang) [inline crates/phactory/Cargo.toml:41]: But the title says 0.9.37 ？🤔
Did you forget to push some further commits?

Comment (jasl) [reply crates/phactory/Cargo.toml:41]: weird...

Comment (jasl) [reply crates/phactory/Cargo.toml:41]: ah push failed...

**Review by kvinwang — APPROVED** [review]
LGTM

**Pull request merged**